tests/testthat/test-utils.R

```diff
@@ -1,18 +1,25 @@
 context("utils.R")
 
 library("synapser")
-if (on_travis()) syn_travis_login() else synLogin()
+attempt_login()
 
 test_that("on_travis() returns TRUE on Travis", {
   expect_equal(on_travis(), isTRUE(as.logical(Sys.getenv("TRAVIS"))))
 })
 
-test_that("login works on travis", {
+test_that("login works on travis in main repo", {
   ## Lots of other things will fail too if it doesn't, but doesn't hurt to have
   ## a dedicated test
-  if (!on_travis()) {
-    skip("This test should run on Travis only")
-  }
+  skip_if_not(on_travis())
+
+  ## In this one we do need to ensure it only runs for builds on upstream repo,
+  ## not forks
+  owner <- gsub(
+    "(^[^/]+)(.+)", "\\1",
+    Sys.getenv("TRAVIS_PULL_REQUEST_SLUG")
+  )
+  skip_if_not(owner == "Sage-Bionetworks", "Testing on upstream repo")
```

Comment:
Only line with explanation of why test is skipped.  Did you want the others have some context like:
```skip_if_not(logged_in(), "Not Logged In")```?  

Not a big deal. Looks good!

Reply:
Mentioned this in slack but I'll put it here too for posterity: there’s some default text that will appear, so they only need a message if you want to override that. skip_if_not(logged_in()) will show "logged_in() is not TRUE" when it’s skipped — I thought that was sufficient in most cases